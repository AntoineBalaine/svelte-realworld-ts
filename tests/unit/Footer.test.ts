import { render } from "@testing-library/svelte";
import { describe, test, expect } from "vitest";
import Footer from "src/components/Footer.svelte";
import type { ComponentProps } from "svelte";

describe("Footer Component", () => {
	test("should render the component", () => {
		const props: ComponentProps<Footer> = {};
		const footerPreview = render(Footer, props);
		expect(footerPreview.getByTestId("footer-component")).not.toBeUndefined();

		const allLinks = footerPreview.getAllByRole("link");
		expect(allLinks[0].innerHTML).toEqual("conduit");
		expect(allLinks[1].innerHTML).toEqual("Thinkster");

		expect(footerPreview.container).toMatchSnapshot();
	});
});
