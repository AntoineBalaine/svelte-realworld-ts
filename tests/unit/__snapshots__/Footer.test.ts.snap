// Vitest Snapshot v1

exports[`Footer Component > should render the component 1`] = `
<body>
  <div>
    <footer
      data-testid="footer-component"
    >
      <div
        class="container"
      >
        <a
          class="logo-font"
          href="/"
        >
          conduit
        </a>
         
        <span
          class="attribution"
        >
          An interactive learning project from 
          <a
            href="https://thinkster.io"
          >
            Thinkster
          </a>
          . Code &
			design licensed under MIT.
        </span>
      </div>
    </footer>
    <!--&lt;Footer&gt;-->
  </div>
</body>
`;
